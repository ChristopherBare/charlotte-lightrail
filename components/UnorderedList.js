import React from 'react';
import { Text, View } from 'react-native';

const UnorderedList = ({ content, styles }) => (
  <View>
    {content.map((text, index) => (
      <View style={styles.bulletWrapper} key={`bullet-${index}`}>
        <View style={styles.bullet}>
          <Text allowFontScaling={false} style={styles.content}>&#9679;</Text>
        </View>
        <View style={styles.bulletContent}>
          <Text allowFontScaling={false} style={styles.content}>{text}</Text>
        </View>
      </View>
    ))}
  </View>
);

UnorderedList.propTypes = {
  content: React.PropTypes.array.isRequired,
  styles: React.PropTypes.object.isRequired,
};

export default UnorderedList;
