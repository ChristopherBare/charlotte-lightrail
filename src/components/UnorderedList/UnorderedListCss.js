import styled from 'styled-components/native';
import { COLORS } from '../../assets/styles/constants';

export const View = styled.View`
  ${props => props.bulletWrapper && "flexDirection: row;"}
  ${props => props.bullet && "width: 13px;"}
  ${props => props.bulletContent && "flex: 1px;"}
`;

export const Text = styled.Text`
  fontSize: 14px;
  color: ${COLORS.primaryTextColor};
`;
