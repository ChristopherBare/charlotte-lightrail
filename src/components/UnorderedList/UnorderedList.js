import React from 'react';
import PropTypes from 'prop-types';
import { Text, View } from './UnorderedListCss';

export default class UnorderedList extends React.Component {

  static propTypes = {
    content: PropTypes.array.isRequired
  }

  render() {
    const { content } = this.props;
    return (
      <View>
        {content.map((text, index) => (
          <View bulletWrapper key={`bullet-${index}`}>
            <View bullet>
              <Text allowFontScaling={false}>&#9679;</Text>
            </View>
            <View bulletContent>
              <Text allowFontScaling={false}>{text}</Text>
            </View>
          </View>
        ))}
      </View>
    );
  }
}
