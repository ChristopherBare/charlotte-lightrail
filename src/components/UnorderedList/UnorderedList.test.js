import 'react-native';
import React from 'react';
import UnorderedList from './UnorderedList';
import renderer from 'react-test-renderer';

describe('<UnorderedList />', () => {
  let props;

  beforeAll(() => {
    props = {
      content: [
        'Find the closest station',
        'Get directions, on foot, or by car',
        'See when the next train is coming',
        'See the full rail schedule',
        'See these FAQs about riding the rails'
      ]
    }
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <UnorderedList {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});