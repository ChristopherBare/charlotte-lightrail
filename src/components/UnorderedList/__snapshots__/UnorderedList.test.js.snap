exports[`<UnorderedList /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }>
  <View
    bulletWrapper={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      bullet={true}
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      bulletContent={true}
      style={
        Array [
          Object {
            "flexBasis": 1,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        Find the closest station
      </Text>
    </View>
  </View>
  <View
    bulletWrapper={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      bullet={true}
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      bulletContent={true}
      style={
        Array [
          Object {
            "flexBasis": 1,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        Get directions, on foot, or by car
      </Text>
    </View>
  </View>
  <View
    bulletWrapper={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      bullet={true}
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      bulletContent={true}
      style={
        Array [
          Object {
            "flexBasis": 1,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See when the next train is coming
      </Text>
    </View>
  </View>
  <View
    bulletWrapper={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      bullet={true}
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      bulletContent={true}
      style={
        Array [
          Object {
            "flexBasis": 1,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See the full rail schedule
      </Text>
    </View>
  </View>
  <View
    bulletWrapper={true}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }>
    <View
      bullet={true}
      style={
        Array [
          Object {
            "width": 13,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        ●
      </Text>
    </View>
    <View
      bulletContent={true}
      style={
        Array [
          Object {
            "flexBasis": 1,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 14,
            },
            undefined,
          ]
        }>
        See these FAQs about riding the rails
      </Text>
    </View>
  </View>
</View>
`;
